tests/all/pooling_allocator.rs

```diff
@@ -44,10 +44,18 @@ fn memory_limit() -> Result<()> {
     // the configured limit
     match Module::new(&engine, r#"(module (memory 4))"#) {
         Ok(_) => panic!("module instantiation should fail"),
-        Err(e) =>
-            e.assert_contains(
-            "memory index 0 has a minimum byte size of 262144 which exceeds the limit of 0x30000 bytes",
-        ),
+        Err(e) => {
+            let msg = format!("{e:?}");
+            eprintln!("Got error: {msg}");
+            assert!(msg.contains(
+                "memory index 0 is unsupported in this pooling allocator \
+                 configuration"
+            ));
+            assert!(msg.contains(
+                "memory has a minimum byte size of 262144 which exceeds \
+                 the limit of 0x30000 bytes"
+            ));
+        }
```

Comment:
The `ErrorExt` trait I added in a PR yesterday means I think we can avoid this refactoring and/or you can use `assert_contains` instead of manually doing `e:?`